recipes/folly/all/test_package/conanfile.py

```diff
@@ -1,18 +1,31 @@
-from conans import ConanFile, CMake, tools
 import os
+from conan import ConanFile
+from conan.tools.cmake import CMake, CMakeToolchain
+from conan.tools.build import can_run
+from conan.tools.cmake import cmake_layout
 
+required_conan_version = ">=1.43.0"
 
 class TestPackageConan(ConanFile):
     settings = "os", "compiler", "build_type", "arch"
-    generators = "cmake", "cmake_find_package_multi"
+    generators = "CMakeDeps", "VirtualRunEnv"
+
+    def requirements(self):
+        self.requires(self.tested_reference_str)
+
+    def generate(self):
+        tc = CMakeToolchain(self)
+        tc.variables["FOLLY_VERSION"] = self.dependencies["fmt"].ref.version
```

Comment:
```suggestion
```
I don't see any usage in test_package/CMakeLists.txt. Also, you tried to capture `fmt` version, which should be `folly` instead. Conan 2 is a bit tricky when exposing transitive dependencies, it only exposes what you explicitly added as requirement.

Reply:
Yup I think I forgot to add this in `test_package/CMakeLists.txt`

```
if (${FOLLY_VERSION} VERSION_LESS "2021.07.20.00")
    set_property(TARGET ${PROJECT_NAME} PROPERTY CXX_STANDARD 14)
else()
    set_property(TARGET ${PROJECT_NAME} PROPERTY CXX_STANDARD 17)
endif()
```

> Also, you tried to capture fmt version, which should be folly instead

uff ctl+c/v mishap... I will change it.

Reply:
Ah no, the code was there but github doesn't show the 'unchanged lines' until you click the "Expand down" button (fail :-P).

I have fixed the self.dependencies .. should be working now.